src/parser/mod.rs

```diff
@@ -7273,6 +7273,10 @@ impl<'a> Parser<'a> {
                 let if_exists = self.parse_keywords(&[Keyword::IF, Keyword::EXISTS]);
                 let name = self.parse_identifier(false)?;
                 AlterTableOperation::DropProjection { if_exists, name }
+            } else if dialect_of!(self is SnowflakeDialect | GenericDialect)
+                && self.parse_keywords(&[Keyword::CLUSTERING, Keyword::KEY])
```

Comment:
for the gating can we use a method on the dialect trait in place of the `dialect_of!`? I'm thinking as well if it would make sense to one method to represent these options since they belong to the same feature , e.g. `supports_alter_table_clustering_action()` or similar method?

Reply:
Perhaps we should drop the gate altogether in this case and introduce gating as necessary? As of now, the gate will only help reject statements the dialect shouldn't accept, vs. parsing incorrectly. Just a thought though...

Reply:
Yeah i think that sounds reasonable!